tests/test_core.py

```diff
@@ -8816,10 +8817,13 @@ def test_undefined_main(self):
     # building library code that has no main.
     # TODO(sbc): Simplify the code by making this an opt-in feature.
     # https://github.com/emscripten-core/emscripten/issues/9640
+
     src = '''
     #include <emscripten.h>
     EMSCRIPTEN_KEEPALIVE void foo() {}
     '''
+    # Clear out the default tests settings since they include IGNORE_MISSING_MAIN=0
+    self.settings_mods = {}
```

Comment:
I've never seen us do this in a test before, it seems wrong to me. That's the internal state, isn't it? Why not modify the state using `set_setting`?

Reply:
Revert all this stuff.